src/tools.ts

```diff
@@ -13,6 +13,67 @@ function isPasteAction (handlerId: string, payload: unknown): payload is PastePa
   return handlerId === 'paste'
 }
 
+function getRangesFromDecorations (
+  editor: monaco.editor.ICodeEditor,
+  decorationFilter: (decoration: monaco.editor.IModelDecoration) => boolean
+): monaco.Range[] {
+  const model = editor.getModel()
+  if (model == null) {
+    return []
+  }
+
+  return model
+    .getAllDecorations()
+    .filter(decorationFilter)
+    .map((decoration) => decoration.range)
+}
+
+/**
+ * Exctract ranges between startToken and endToken
+ */
+export function extractRangesFromTokens (editor: monaco.editor.ICodeEditor, startToken: string, endToken: string, isRegex: boolean = false): monaco.Range[] {
+  const editorModel = editor.getModel()
+  const ranges: monaco.Range[] = []
+  if (editorModel != null) {
+    let currentPosition = editorModel.getFullModelRange().getStartPosition()
+    let match: monaco.editor.FindMatch | null
+    while ((match = editorModel.findNextMatch(startToken,
+      /* searchStart */currentPosition,
+      /* isRegex */isRegex,
+      /* matchCase */true,
+      /* wordSeparators */null,
+      /* captureMatches */false
+    )) != null) {
+      if (match.range.getStartPosition().isBefore(currentPosition)) {
+        break
+      }
+      const matchEnd = editorModel.findNextMatch(endToken,
+        /* searchStart */match.range.getEndPosition(),
+        /* isRegex */isRegex,
+        /* matchCase */true,
+        /* wordSeparators */null,
+        /* captureMatches */false
+      )
+      if (matchEnd != null && matchEnd.range.getStartPosition().isBefore(match.range.getStartPosition())) {
+        break
+      }
+      currentPosition = matchEnd?.range.getEndPosition() ?? editorModel.getFullModelRange().getEndPosition()
+      ranges.push(monaco.Range.fromPositions(match.range.getStartPosition(), currentPosition))
+    }
+    return ranges
+  }
+  return []
+}
+
+export function createDecorationsForRanges (
+  editor: monaco.editor.ICodeEditor,
+  ranges: monaco.IRange[],
+  options: monaco.editor.IModelDecorationOptions
+): monaco.editor.IEditorDecorationsCollection {
+  const decorations = ranges.map((range) => ({ options, range }))
+  return editor.createDecorationsCollection(decorations)
+}
```

Comment:
are you sure it's worth it to have this method? :thinking:

Reply:
I can remove it from the lib and have the logic directly in the IDE, it only needed once in the IDE.

Reply:
24a86fd